model/event/subscriptions.go

```diff
@@ -238,14 +251,29 @@ func (s *Subscription) String() string {
 	return out
 }
 
-func FindSubscriptionsByOwner(owner string) ([]Subscription, error) {
+func FindSubscriptionsByOwner(owner string, ownerType OwnerType) ([]Subscription, error) {
 	if len(owner) == 0 {
 		return nil, nil
 	}
+	if !IsValidOwnerType(string(ownerType)) {
+		return nil, errors.Errorf("%s is not a valid owner type", ownerType)
+	}
 	query := db.Query(bson.M{
-		subscriptionOwnerKey: owner,
+		subscriptionOwnerKey:     owner,
+		subscriptionOwnerTypeKey: ownerType,
 	})
 	subscriptions := []Subscription{}
 	err := db.FindAllQ(SubscriptionsCollection, query, &subscriptions)
 	return subscriptions, errors.Wrapf(err, "error retrieving subscriptions for owner %s", owner)
 }
+
+func IsValidOwnerType(in string) bool {
```

Comment:
This should be a method on the OwnerType type.

Reply:
I'm using this to validate input before converting to an OwnerType, so this needs to have its current signature